.github/workflows/ci-code.yml

```diff
@@ -2,7 +2,7 @@ name: continuous-integration-code
 
 on:
   push:
-    branches-ignore: [gh-pages]
```

Comment:
Is this intentional? Now the normal unit tests workflow will only run on merge to main right and no longer on push to branches?

Reply:
The unit tests will still run on pull_request event. 

I made this change for this PR in particular since I was opening the pull request from origin, and so the status checks were all duplicated. This is what happens for PRs from forks as well, essentially wasting CI compute imo. 

So this change would only make a difference to you if you care about running tests on push before you open your PR. Given that you can always open a draft PR I feel like this use case is not as important, but I am happy to revert this in case you think differently.

Thanks for spotting this! I totally forgot I did this here. Sorry about that.

Reply:
I agree that in most cases, having it run doubly is not ideal.

> So this change would only make a difference to you if you care about running tests on push before you open your PR. Given that you can always open a draft PR

This is the critical point for me though. In many cases, I rely on running the CI on my fork. I don't really like the concept of draft PRs and so I rarely open them. If the work is not ready, there is rarely a reason to open a (draft) PR. The downside for me as a reviewer is that it adds a lot of noise. I look at the PR page often to see what needs to be done and when. Having a lot of open draft PRs where everytime I have to assess whether to consider or skip it, adds noise. So I don't really like the idea that users will be forced to open them just to get the CI to run.

I could be convinced otherwise, but I think this discussion shows that we probably shouldn't sneak the change into this PR, so for now please remove it and I will merge

Reply:
No problem, done.